python-threatexchange/threatexchange/non_threatexchange_api.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import typing as t
+from .api_representations import HashRecord
+from .api import ThreatExchangeAPI
+
+
+class NonThreatExchangeAPI(ThreatExchangeAPI):
```

Comment:
This feels very clunky. Why is `NonThreatExchangeAPI ` a subclass of `ThreatExchangeAPI`? Feels semantically disconnected.

I understand that this is what you're trying to do:
```
python-threatexchange

ThreatExchangeAPI ----> NonThreatExchangeAPI

hmalib 

Fetcher uses a class of ThreatExchangeAPI
```

But this will very quickly run into issues. Because NonThreatExchangeAPI does not belong in python-threatexchange.

Instead, how about this:

Let's not touch ThreatExchangeAPI at all. Imagine the OSS hash-exchange provided its own python API package and you had to create a middle layer to abstract `python-threatexchange` and the `oss-hash-exchange-api` package.

You'd need to do that in hmalib.

Reply:
I was trying to do this in hmalib first, but then I found there are lots of common methods I can use in ThreatExchangeAPI, so that's why I put both them in the same place. But you're right, it's confusing! I will create a common API class in hmalib so that both ThreatExchangeAPI and NonThreatExchangeAPI can extend it.